redbot/core/events.py

```diff
@@ -241,10 +239,112 @@ async def on_command_error(ctx, error, unhandled_by_cog=False):
+                elif error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be an integer no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Argument `{parameter_name}` must be an integer no less than {maximum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be an integer between {minimum} and {maximum}."
+                    )
+            elif isinstance(error.value, float):
+                if (error.minimum == 0 or error.minimum == 1) and error.maximum is None:
+                    message = _("Argument `{parameter_name}` must be a positive number.")
+                elif error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number no less than {maximum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a number between {minimum} and {maximum}."
+                    )
+            else:
+                if error.minimum is None and error.maximum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of no more than {maximum}."
+                    )
+                elif error.minimum is not None and error.maximum is None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of no less than {minimum}."
+                    )
+                elif error.maximum is not None and error.minimum is not None:
+                    message = _(
+                        "Argument `{parameter_name}` must be a string with a length of between {minimum} and {maximum}."
+                    )
+            await ctx.send(
+                message.format(
+                    maximum=error.maximum,
+                    minimum=error.minimum,
+                    parameter_name=ctx.current_parameter.name,
+                )
+            )
+            return
         elif isinstance(error, commands.BadArgument):
+            if isinstance(converter, commands.Range):
+                if converter.annotation is int:
+                    if (converter.min == 0 or converter.min == 1) and converter.max is None:
```

Comment:
Whilst this check makes sense, if the converter minimum were to be 1, and the user passes "0" to the converter, the message would still ask for a positive integer despite zero being a positive integer. Is the `== 1` check really needed?

![image](https://user-images.githubusercontent.com/67752638/232164954-13ddfffd-4b4b-457a-923b-a9cb4572729a.png)